### ISSUE TITLE
MSVC fails 1.86.0 process.vs with definition of dllimport function not allowed

### ISSUE THREAD

**Issue by bebuch** [boostorg/process#408]
```
boost/process/v2/detail/utf8.hpp(28): error C2491: 'boost::process::v2::detail::conv_string': definition of dllimport function not allowed
```

In this file are 3 template functions defined as dllimport/dllexport which is invalid.

**Comment by Tectu** [comment]
I'm observing the same issue with MSYS2/GCC and boost 1.86.0:
```
C:/msys64/mingw64/include/boost/process/v2/detail/utf8.hpp:28:47: error: function 'std::__cxx11::basic_string<_CharT, _Traits, _Allocator> boost::process::v2::detail::conv_string(const CharIn*, std::size_t, Allocator)' definition is marked dllimport
   28 | std::basic_string<CharOut, Traits, Allocator> conv_string(
      |                                               ^~~~~~~~~~~
C:/msys64/mingw64/include/boost/process/v2/detail/utf8.hpp:40:47: error: function 'std::__cxx11::basic_string<_CharT, _Traits, _Allocator> boost::process::v2::detail::conv_string(const wchar_t*, std::size_t, Allocator)' definition is marked dllimport
   40 | std::basic_string<CharOut, Traits, Allocator> conv_string(
      |                                               ^~~~~~~~~~~
C:/msys64/mingw64/include/boost/process/v2/detail/utf8.hpp:64:47: error: function 'std::__cxx11::basic_string<_CharT, _Traits, _Allocator> boost::process::v2::detail::conv_string(const char*, std::size_t, Allocator)' definition is marked dllimport
   64 | std::basic_string<CharOut, Traits, Allocator> conv_string(
```

**Comment by bebuch** [comment]
This was fixed via #409.

**Comment by Tectu** [comment]
I can confirm that boost 1.87.0 no longer suffers from this issue.
